File: src/database/supabase/storage/profileStorage.js
import { supabase, BUCKETS } from '../config';

/**
 * UPLOAD PROFILE PICTURE
 */
export const uploadProfilePicture = async (userId, file) => {
  try {
    const fileName = `${userId}/profile.jpg`;
    
    const { data, error } = await supabase.storage
      .from(BUCKETS.USER_PROFILES)
      .upload(fileName, file, {
        cacheControl: '3600',
        upsert: true
      });

    if (error) throw error;

    const { data: publicUrlData } = supabase.storage
      .from(BUCKETS.USER_PROFILES)
      .getPublicUrl(fileName);

    return { 
      success: true, 
      url: publicUrlData.publicUrl 
    };
  } catch (error) {
    return { 
      success: false, 
      error: error.message 
    };
  }
};

/**
 * DELETE PROFILE PICTURE
 */
export const deleteProfilePicture = async (userId) => {
  try {
    const fileName = `${userId}/profile.jpg`;
    
    const { error } = await supabase.storage
      .from(BUCKETS.USER_PROFILES)
      .remove([fileName]);

    if (error) throw error;

    return { success: true };
  } catch (error) {
    return { 
      success: false, 
      error: error.message 
    };
  }
};

File: app/src/main/java/com/example/bunnix/database/supabase/storage/ProfileStorage.kt
package com.example.bunnix.database.supabase.storage

import android.content.Context
import android.net.Uri
import com.example.bunnix.database.config.SupabaseConfig
import io.github.jan.supabase.storage.storage
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import java.io.File

object ProfileStorage {

    private val storage = SupabaseConfig.client.storage

    // UPLOAD PROFILE PICTURE
    suspend fun uploadProfilePicture(
        context: Context,
        userId: String,
        imageUri: Uri
    ): Result<String> = withContext(Dispatchers.IO) {
        try {
            val file = uriToFile(context, imageUri)
            val fileBytes = file.readBytes()
            val path = "$userId/profile.jpg"

            val bucket = storage.from(SupabaseConfig.Buckets.USER_PROFILES)

            // ✅ CORRECT: Using options builder
            bucket.upload(
                path = path,
                data = fileBytes,
                upsert = true
            )

            val publicUrl = bucket.publicUrl(path)
            file.delete()

            Result.success(publicUrl)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    // UPLOAD VENDOR COVER
    suspend fun uploadVendorCover(
        context: Context,
        vendorId: String,
        imageUri: Uri
    ): Result<String> = withContext(Dispatchers.IO) {
        try {
            val file = uriToFile(context, imageUri)
            val fileBytes = file.readBytes()
            val path = "$vendorId/cover.jpg"

            val bucket = storage.from(SupabaseConfig.Buckets.VENDOR_PHOTOS)

            // ✅ CORRECT: Using options builder
            bucket.upload(
                path = path,
                data = fileBytes,
                upsert = true
            )

            val publicUrl = bucket.publicUrl(path)
            file.delete()

            Result.success(publicUrl)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    private fun uriToFile(context: Context, uri: Uri): File {
        val inputStream = context.contentResolver.openInputStream(uri)
            ?: throw Exception("Cannot open URI")

        val tempFile = File(context.cacheDir, "temp_${System.currentTimeMillis()}.jpg")
        tempFile.outputStream().use { output ->
            inputStream.copyTo(output)
        }
        inputStream.close()

        return tempFile
    }
}